contracts/src/CrossChainApplications/GETTING_STARTED.md

```diff
@@ -20,245 +20,277 @@ pragma solidity 0.8.18;
 
 ```solidity
-// Send a new message to another chain.
-function sendMessage(
-    bytes32 destinationBlockchainID,
-    address destinationAddress,
-    address feeTokenAddress,
-    uint256 feeAmount,
-    uint256 requiredGasLimit,
-    string calldata message
-) external returns (bytes32 messageID) {}
+    /**
+     * @dev Send a new message to another chain.
+     */
+    function sendMessage(
+        bytes32 destinationBlockchainID,
+        address destinationAddress,
+        address feeTokenAddress,
+        uint256 feeAmount,
+        uint256 requiredGasLimit,
+        string calldata message
+    ) external returns (bytes32 messageID) {
+
+    }
 ```
 
-`MyExampleCrossChainMessenger` also needs to implement `ITeleporterReceiver` by adding the method `receiveTeleporterMessage` that receives the cross-chain messages from Teleporter.
+`MyExampleCrossChainMessenger` also needs to implement `ITeleporterReceiver`. First, add the import of this interface:
 
 ```solidity
-// Receive a new message from another chain.
-function receiveTeleporterMessage(
-    bytes32 sourceBlockchainID,
-    address originSenderAddress,
-    bytes calldata message
-) external {}
+import {ITeleporterReceiver} from "@teleporter/ITeleporterReceiver.sol";
+```
+
+Then declare that the contract will implement it:
+
+```diff
+  contract MyExampleCrossChainMessenger is
+-     ReentrancyGuard
++     ReentrancyGuard,
++     ITeleporterReceiver
+  {
+```
+
+And then finally add the method `receiveTeleporterMessage` that receives the cross-chain messages from Teleporter.
+
+```solidity
+    /**
+     * @dev Receive a new message from another chain.
+     */
+    function receiveTeleporterMessage(
+        bytes32 sourceBlockchainID,
+        address originSenderAddress,
+        bytes calldata message
+    ) external {
+
+    }
 ```
 
 Now it's time to implement the methods, starting with `sendMessage`. First, add the necessary imports.
 
 ```solidity
-import {
-    ITeleporterMessenger,
-    TeleporterMessageInput,
-    TeleporterFeeInfo
-} from "@teleporter/ITeleporterMessenger.sol";
 import {SafeERC20TransferFrom, SafeERC20} from "@teleporter/SafeERC20TransferFrom.sol";
-import {ITeleporterReceiver} from "@teleporter/ITeleporterReceiver.sol";
 import {IERC20} from "@openzeppelin/contracts@4.8.1/token/ERC20/IERC20.sol";
-import {ReentrancyGuard} from "@openzeppelin/contracts@4.8.1/security/ReentrancyGuard.sol";
 ```
 
 Next, add a `using` directive in the contract declaration to specify `SafeERC20` as the `IERC20` implementation to use:
 
 ```solidity
-contract ExampleCrossChainMessenger is
+contract MyExampleCrossChainMessenger is
     ReentrancyGuard,
     ITeleporterReceiver
 {
+```
+```solidity
     using SafeERC20 for IERC20;
+```
+```solidity
```

Comment:
The code blocks appear to be off here.

Reply:
the code blocks are "split" such that the bit that's to be copied for this particular instruction is in its own code block, whereas the surrounding "context" code, which is repeated from earlier instructions, is in is not included in the code block that needs to be copied to follow the instruction. this way the reader can easily hit the "copy" button on the code block needed to follow the instruction, without also copying the "context code" that they have already copied and pasted while following previous instructions.

i also explained that in the commit message [here](https://github.com/ava-labs/teleporter/pull/337/commits/fb8c1c045bcf61c3f136a5e26b8268d2605f6664). but i wrote the explanation for this comment fresh off the top of my head. i know the logic is a little complicated/convoluted, but hopefully _one_ of these explanations helps convey what i'm getting at. :)